app/models/service.rb

```diff
@@ -521,6 +521,18 @@ def can_use_backends?
     proxy.apicast_configuration_driven && !proxy.service_mesh_integration?
   end
 
+  # TODO: Remove this when no one use plugins
+  def plugin_deployment?
+    deployment = self[:deployment_option]
+    DeploymentOption.plugins.include?(deployment)
+  end
+
+  # TODO: Remove this when no one use plugins
+  def deployment_option
+    return 'hosted' if plugin_deployment?
```

Comment:
I think it should be self_managed
People using plugin do not really have any apicast in front. It would enforce them to have one hosted
Also zync will create a route automatically in OpenShift if it is hosted

Reply:
I’d say the logical setting is what you get out of the box when you install 3scale. That would be 3scale managed. Other question is if it would work.

Reply:
Yes, everything is shown as if 3scale managed APIcast was being used. And when the customer clicks ok, setting will actually be updated in db. I agree this is not the final solution but it is kind of a nice in between with heads up to customer. Next step is remove all of it.

Reply:
We can remove the code later but a data migration (DML) should be done IMO
Anyway plugins are not shown anymore right?

Reply:
They are still shown (I just verified), deprecation notice was already issued long time ago. This is adding more work and has really unknown behaviours

Also this undesirable behaviour is worse than leaving it there: Zync will just create routes if it is hosted unlike before while it was still plugin.

If we just display the message and not touch the `deployment_option` method I would merge the PR.

Reply:
Why is it wrong that zync creates the routes? If they change to apicast because they want to that would also happen, no?

Reply:
It is a change of behaviour without notice zync will create tons of routes ...
Not to mention that some users reported that already https://github.com/3scale/zync/issues/236

If they want to use the hosted gateway yes, but they might also want to use the self managed gateway or the service mesh.

Agree that it makes sense to have default to hosted but I'd say the timing is pretty bad to take this decision ...